Sources/MsbBitWriter.swift

```diff
@@ -54,6 +54,22 @@ public final class MsbBitWriter: BitWriter {
             mask >>= 1
         }
     }
+    
+    /**
+     Writes `number`, using and advancing by `bitsCount` BIT positions.
+
+     - Note: If `bitsCount` is smaller than the actual amount of `number`'s bits than the `number` will be truncated to
+     fit into `bitsCount` amount of bits.
+     - Note: Bits of `number` are processed using the same bit-numbering scheme as of the writer (i.e. "MSB 0").
+     - Note: this method is specifically useful when needing to write a UInt64 which can overflow and crash if converting to an Int when using the regular `write` method
+     */
+    public func write(unsignedNumber: UInt, bitsCount: Int) {
+        var mask: UInt = 1 << (UInt(bitsCount) - 1)
```

Comment:
I suggest to use `UInt(truncatingIfNeeded:)` instead of `UInt()`. The reason for this is that some time ago I observed that integer initializers with truncatingIfNeeded argument label are noticeably faster in terms of compilation times (for some reason, they were type-checked faster). It is possible that this issue was fixed in the most recent versions of Swift, but I am not sure.